spark/src/main/scala/io/delta/sharing/spark/DeltaSharingClient.scala

```diff
@@ -183,13 +192,14 @@ private[spark] class DeltaSharingRestClient(
   override def getFiles(
       table: Table,
       predicates: Seq[String],
-      limit: Option[Long]): DeltaTableFiles = {
+      limit: Option[Long],
+      versionOf: Option[Long]): DeltaTableFiles = {
     val encodedShareName = URLEncoder.encode(table.share, "UTF-8")
     val encodedSchemaName = URLEncoder.encode(table.schema, "UTF-8")
     val encodedTableName = URLEncoder.encode(table.name, "UTF-8")
     val target = getTargetUrl(
       s"/shares/$encodedShareName/schemas/$encodedSchemaName/tables/$encodedTableName/query")
-    val (version, lines) = getNDJson(target, QueryTableRequest(predicates, limit))
+    val (version, lines) = getNDJson(target, QueryTableRequest(predicates, limit, versionOf))
```

Comment:
Should we add: 
  require(versionOf.isEmpty || versionOf.get == version) ?

Reply:
added.